### ISSUE TITLE
FURB118 static method "false positive"

### ISSUE THREAD

**Issue by Rhinopitheque** [astral-sh/ruff#11045]
The rule FURB118 detects "false positives" on classes methods.

Keywords: "FURB118", "method"
```
@staticmethod 
    def get_postal_code(area_code: str) -> str:
        return area_code[:5]
```
Ruff v0.4.0 used with pre-commit hook:
```
  - repo: https://github.com/charliermarsh/ruff-pre-commit
    rev: v0.4.0
    hooks:
      - id: ruff
```

**Comment by charliermarsh** [comment]
Can you share more on why `FURB118` doesn't apply here?

**Comment by Rhinopitheque** [comment]
Let's say we have this class:
```python
class Foo:
    ...
    @staticmethod
    def get_postal_code(area_code: str) -> str:
        return area_code[:5]
```
If I understand the rule correctly, the suggested change would be this I suppose?
```diff
-Foo.get_postal_code("AAAAABB")
+operator.itemgetter(slice(5))("AAAAABB")
```
**So this is not really a false positive but more than removing the `staticmethod` altogether makes us lose a lot of context.**
Usually when people define `staticmethod` like that it's because the behavior is somehow linked to the class.
I feel like this rule should maybe not trigger on static method ?

I'm not so sure, feel free to close.

(Also I feel like `operator.itemgetter(slice(...))` is not really readable, I'dl rather define a lambda for these cases)

Anyway, thanks for the tool and the quick response!

PS: This is a bit related to #10898
